docs/src/man/subduction2D/subduction2D.md

```diff
@@ -10,238 +10,255 @@ We will use GeophysicalModelGenerator.jl to generate the initial geometry, mater
+        nout    = 1e2,
+        verbose = true,
     )
 )
-```
-2. calculate the second invariant and push to history vectors
-```julia
-tensor_invariant!(stokes.ε)
-push!(τII, maximum(stokes.τ.xx))
-
-@parallel (@idx ni .+ 1) multi_copy!(@tensor(stokes.τ_o), @tensor(stokes.τ))
-@parallel (@idx ni) multi_copy!(
-    @tensor_center(stokes.τ_o), @tensor_center(stokes.τ)
+subgrid_characteristic_time!(
+    subgrid_arrays, particles, dt₀, phase_ratios, rheology_augmented, thermal, stokes, xci, di
+)
+centroid2particle!(subgrid_arrays.dt₀, xci, dt₀, particles)
+subgrid_diffusion!(
+    pT, thermal.T, thermal.ΔT, subgrid_arrays, particles, xvi,  di, dt
 )
-
-it += 1
-t  += dt
-
-push!(sol, solution(εbg, t, G0, η0))
-push!(ttot, t)
 ```
-# Visualization
-We will use `Makie.jl` to visualize the results
+
+5. Particles advection
 ```julia
-using GLMakie
+# advect particles in space
+advection!(particles, RungeKutta2(), @velocity(stokes), grid_vxi, dt)
+# advect particles in memory
+move_particles!(particles, xvi, particle_args)
+# check if we need to inject particles
+inject_particles_phase!(particles, pPhases, (pT, ), (T_buffer, ), xvi)
+# update phase ratios
+phase_ratios_center!(phase_ratios, particles, grid, pPhases)
 ```
 
-## Fields
+6. **Optional:** Save data as VTK to visualize it later with [ParaView](https://www.paraview.org/)
 ```julia
- # visualisation of high density inclusion
-th    = 0:pi/50:3*pi;
-xunit = @. radius * cos(th) + 0.5;
-yunit = @. radius * sin(th) + 0.5;
-
-fig   = Figure(size = (1600, 1600), title = "t = $t")
-ax1   = Axis(fig[1,1], aspect = 1, title = L"\tau_{II}", titlesize=35)
-ax2   = Axis(fig[2,1], aspect = 1, title = L"E_{II}", titlesize=35)
-ax3   = Axis(fig[1,2], aspect = 1, title = L"\log_{10}(\varepsilon_{II})", titlesize=35)
-ax4   = Axis(fig[2,2], aspect = 1)
-heatmap!(ax1, xci..., Array(stokes.τ.II) , colormap=:batlow)
-heatmap!(ax2, xci..., Array(log10.(stokes.EII_pl)) , colormap=:batlow)
-heatmap!(ax3, xci..., Array(log10.(stokes.ε.II)) , colormap=:batlow)
-lines!(ax2, xunit, yunit, color = :black, linewidth = 5)
-lines!(ax4, ttot, τII, color = :black)
-lines!(ax4, ttot, sol, color = :red)
-hidexdecorations!(ax1)
-hidexdecorations!(ax3)
-save(joinpath(figdir, "$(it).png"), fig)
-fig
+Vx_v = @zeros(ni.+1...)
+Vy_v = @zeros(ni.+1...)
+velocity2vertex!(Vx_v, Vy_v, @velocity(stokes)...) # interpolate velocity from staggered grid to vertices
+data_v = (; # data @ vertices
+    T   = Array(T_buffer),
+    τII = Array(stokes.τ.II),
+    εII = Array(stokes.ε.II),
+    Vx  = Array(Vx_v),
+    Vy  = Array(Vy_v),
+)
+data_c = (; # data @ centers
+    P   = Array(stokes.P),
+    η   = Array(stokes.viscosity.η_vep),
+)
+velocity_v = ( # velocity vector field
+    Array(Vx_v),
+    Array(Vy_v),
+)
+save_vtk(
+    joinpath(@__DIR__, "vtk_" * lpad("$it", 6, "0")),
+    xvi,
+    xci,
+    data_v,
+    data_c,
+    velocity_v
+)
 ```
-
+      
 ### Final model
+Solution after 990 time steps
 ![](990.png)
```

Comment:
The 990.png needs to be pushed to assets and linked properly. Also maybe rename it to subduction2D_990.png. Same thing for the other figures.
```suggestion
![](../assets/subduction2D_990.png)
```